### PR TITLE
Pyband: Refactor and wrote tests for client and send_tx

### PR DESCRIPTION
**Pull request by pzshine**
## Implementation details

- Refactor the client module
  -  Changed the way to send tx to be separated by mode
   - `send_tx_sync_mode`
   - `send_tx_async_mode`
   - `send_tx_block_mode`
- Added new dataclasses on data module
  - `TransactionSyncMode`
  - `TransactionAsyncMode`
  - `TransactionBlockMode`
- Added new test cases for the client api

**Note:** About `get_latest_block`, it will be on the other pr.

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [x] The pull request includes any and all appropriate unit/integration tests
- [x] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: helpers/pyband/pyband/client.py [hunk 1]
```diff
@@ -1,37 +1,58 @@
 import requests
+import json
+from pyband.transaction import Transaction
```
Comment (evilpeach) [inline helpers/pyband/pyband/client.py]: Is it used?

File: helpers/pyband/tests/client/send_tx_test.py [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+                            {
+                                "key": "validator",
+                                "value": "bandvaloper1p40yh3zkmhcv0ecqp3mcazy83sa57rgjde6wec",
+                            },
+                        ],
+                    },
+                ],
+            }
+        ],
+    )
```
Comment (evilpeach) [inline helpers/pyband/tests/client/send_tx_test.py]: add new line

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: helpers/pyband/pyband/data.py [hunk 3]
```diff
@@ -98,3 +98,21 @@ class Account(object):
     account_number: int
     sequence: int
 
+
+@dataclass
+class TransactionSyncMode(object):
+    tx_hash: str
```
Comment (taobun) [inline helpers/pyband/pyband/data.py]: Add code and log.

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by taobun — APPROVED** [review]
LGTM

File: helpers/pyband/pyband/client.py [hunk 4]
```diff
@@ -1,39 +1,73 @@
...
+
+    def send_tx_block_mode(self, data: dict) -> TransactionBlockMode:
+        data = self._post("/txs", json={"tx": data, "mode": "block"})
+        if "code" in data:
+            code = int(data["code"])
+            error_log = data["raw_log"]
+            log = []
+        else:
+            code = 0
+            log = data["logs"]
```
Comment (sorawit) [inline helpers/pyband/pyband/client.py]: Let's make `log` / `error_log` order consistent with the `if` block above

File: helpers/pyband/pyband/data.py [hunk 5]
```diff
@@ -98,3 +98,25 @@ class Account(object):
...
```
Comment (sorawit) [inline helpers/pyband/pyband/data.py]: should `tx_hash` be `str` or `bytes`?

**Review by sorawit — CHANGES_REQUESTED** [review]

**Review by sorawit — APPROVED** [review]
Looks good

**Review by evilpeach — APPROVED** [review]

**Pull request merged**